### PR TITLE
[sw, ottf] Enable chip level tests to boot with mask_rom and rom_ext

### PR DESCRIPTION
**Pull request by engdoreis**
This PR enables the chip level test to boot with mask_rom and rom_ext.
#### Executing in the 2nd stage
```
bazel-bin/sw/host/opentitantool/opentitantool load-bitstream /tmp/bitstream-latest/lowrisc_systems_chip_earlgrey_cw310_0.1.bit.splice

bazel run //sw/host/opentitantool -- --exec "bootstrap /home/doreis/lowrisc/git/opentitan/bazel-bin/sw/device/tests/aes_entropy_test_creator_prog_fpga_cw310.test_key_0.signed.bin" console
```
#### Executing in the 3nd stage
```
bazel-bin/sw/host/opentitantool/opentitantool load-bitstream /tmp/bitstream-latest/lowrisc_systems_chip_earlgrey_cw310_0.1.bit.splice

bazel run //sw/host/opentitantool -- --exec "bootstrap /home/doreis/lowrisc/git/opentitan/bazel-bin/sw/device/silicon_creator/rom_ext/virtual_addr_fpga_cw310.test_key_0.signed.bin@0 /home/doreis/lowrisc/git/opentitan/bazel-bin/sw/device/tests/aes_entropy_test_owner_prog_fpga_cw310.test_key_0.signed.bin@0x10000" console
```

Fixes: #10712

### CODE REVIEW

File: sw/device/lib/testing/test_framework/BUILD [hunk 1]
```diff
@@ -67,7 +84,8 @@ cc_library(
     ],
     target_compatible_with = [OPENTITAN_CPU],
     deps = [
-        ":linker_script",
+        # ":ld_test_rom",
+        ":ld_silicon_creator_stage",
```
Comment (engdoreis) [inline sw/device/lib/testing/test_framework/BUILD:88]: How to address that in Bazel? maybe creating multimple targets?

File: sw/device/tests/BUILD [hunk 2]
```diff
@@ -43,6 +43,7 @@ opentitan_functest(
 opentitan_functest(
     name = "aes_idle_test",
     srcs = ["aes_idle_test.c"],
+    signed = True,
```
Comment (engdoreis) [inline sw/device/tests/BUILD:46]: Make all tests signed?

File: sw/device/tests/BUILD [hunk 3]
```diff
@@ -36,6 +31,39 @@ opentitan_functest(
     ],
 )
 
+opentitan_functest(
```
Comment (engdoreis) [inline sw/device/tests/BUILD:34]: This is probably not the best approach to create these new binaries, should I create an issue for someone with more expertise in Bazel?

**Comment by alphan** [comment]
@engdoreis I'll review this soon but meanwhile can you take a look at the CI failures? Thanks!

**Comment by timothytrippel** [comment]
After discussion with @engdoreis , closing this in favor of #14190 (and follow up PRs).

**Pull request closed without merging**